src/lib/Tests/RepositoryForms/View/ViewParametersListenerTest.php

```diff
@@ -0,0 +1,164 @@
+
+        $locationService = $this->createMock(LocationService::class);
+        $locationService->expects(self::once())
+            ->method('loadParentLocationsForDraftContent')
+            ->with($versionInfo)
+            ->willReturn($locations);
+
+        $viewParametersListener = new ViewParametersListener($locationService);
+
+        $viewParametersListener->setViewTemplateParameters($event);
+
+        $this->assertSame($locations, $contentView->getParameter('parentLocations'));
+    }
+
+    public function testSetViewTemplateParametersWithMainLocationId()
+    {
+        $mainLocationId = 123;
+
+        $contentInfo = $this->generateContentInfo($mainLocationId);
+
+        $versionInfo = $this->generateVersionInfo($contentInfo);
+
+        $contentView = new ContentEditView();
+        $contentView->setParameters([
+            'content' => $this->generateContent($versionInfo),
+            'parentLocations' => [],
+        ]);
+
+        $event = new PreContentViewEvent($contentView);
+
+        $locationService = $this->createMock(LocationService::class);
+        $locationService->expects(self::never())
+            ->method('loadParentLocationsForDraftContent');
+
+        $viewParametersListener = new ViewParametersListener($locationService);
+
+        $viewParametersListener->setViewTemplateParameters($event);
+
+        $this->assertSame([], $contentView->getParameter('parentLocations'));
+    }
+
+    public function testSetViewTemplateParametersWithoutContentEditViewInstance()
+    {
+        $contentView = $this->createMock(View::class);
+
+        $locationService = $this->createMock(LocationService::class);
+        $locationService->expects(self::never())
+            ->method('loadParentLocationsForDraftContent');
+
+        $viewParametersListener = new ViewParametersListener($locationService);
+
+        $this->assertNull($viewParametersListener->setViewTemplateParameters(new PreContentViewEvent($contentView)));
+    }
+
+    public function testSubscribedEvents()
+    {
+        $locationService = $this->createMock(LocationService::class);
+        $locationService->expects(self::never())
+            ->method('loadParentLocationsForDraftContent');
+
+        $viewParametersListener = new ViewParametersListener($locationService);
+
+        $this->assertSame([MVCEvents::PRE_CONTENT_VIEW => 'setViewTemplateParameters'], $viewParametersListener::getSubscribedEvents());
+    }
+
+    /**
+     * @param VersionInfo $versionInfo
+     *
+     * @return ApiContent
+     */
+    private function generateContent(VersionInfo $versionInfo): ApiContent
+    {
+        return new Content(
+            [
+                'versionInfo' => $versionInfo,
+            ]
+        );
+    }
+
+    /**
+     * @param ContentInfo $contentInfo
+     *
+     * @return APIVersionInfo
+     */
+    private function generateVersionInfo(APIContentInfo $contentInfo): APIVersionInfo
+    {
+        return new VersionInfo(
+            [
```

Comment:
For arrays with only one element you can write it inline. No need to expand it as it takes more lines and forces you to scroll more.